fix: ab playback resolver not allowing sessions to span between segments SUPERFLY-5

Julusian opened the pull request:
As discussed in SUPERFLY-5, AB sessions should persist across parts, and it should be possible to do so across segments.

In the code it was doing `Check if we can continue sessions from the part before, or if we should create new ones`, but the check was only looking at the rank of the parts, which is really the rank inside the segment. 

This expands that check to correctly consider the rank of the owning segment when they are in different segments and consider the rank of the parts only when they are in the same segment.

This has been verified with the addition of a new unit test, no manual testing has been performed currently.

Julusian committed:
fix: ab playback resolver not allowing sessions to span between segments

PeterC89 reviewed (approved):
Approved from a functionality perspective

olzzon committed:
Merge remote-tracking branch 'upstream/release51' into fix/ab-spanning-parts

Julusian committed:
feat: allow ab sessions to declare themselves as 'globally unique'

This bypasses any sessionName collision/restarting logic, and lets blueprints take care of that internally

Julusian committed:
Revert "fix: ab playback resolver not allowing sessions to span between segments"

This reverts commit b3b4043b92f37e12823e51cc037672cc50cded4f.

Julusian committed:
chore: unit tests

olzzon committed:
fix: reorder ab assignments if part has been inserted

Julusian committed:
wip: add back ab logging

Julusian committed:
fix: ensure ab sessions get reordered when adlibbing a fixed duration piece

Julusian committed:
Revert "fix: reorder ab assignments if part has been inserted"

This reverts commit 8d528d70188a8af63506cd9de1f25578bc936449.

Julusian committed:
fix: ensure pieces don't claim to have durations when their end has not yet been decided

Julusian commented:
Upstream: https://github.com/nrkno/sofie-core/pull/1261
